numpy/_core/src/multiarray/mapping.h

```diff
@@ -4,6 +4,80 @@
 extern NPY_NO_EXPORT PyMappingMethods array_as_mapping;
 
 
+/*
+ * Store the information needed for fancy-indexing over an array. The
+ * fields are slightly unordered to keep consec, dataptr and subspace
+ * where they were originally.
```

Comment:
Is there any need to reorder the fields?

Reply:
Hmmm, will have a look.  I am not quite sure I found everything that can be cleaned out also.